app/src/sign_tx_context.rs

```diff
@@ -0,0 +1,192 @@
+use ledger_device_sdk::ecc::Secp256k1;
+use ledger_device_sdk::ecc::SeedDerive;
+use ledger_device_sdk::io::ApduHeader;
+use ledger_device_sdk::ui::gadgets::MessageScroller;
+use utils::{buffer::Buffer, decode::PartialDecoder, deserialize_path, types::UnsignedTx};
+
+use crate::blind_signing::is_blind_signing_enabled;
+use crate::nvm_buffer::NVMData;
+use crate::nvm_buffer::NvmBuffer;
+use crate::nvm_buffer::NVM;
+use crate::tx_reviewer::TxReviewer;
+use crate::{
+    blake2b_hasher::{Blake2bHasher, BLAKE2B_HASH_SIZE},
+    error_code::ErrorCode,
+};
+
+const SIZE: usize = 2048;
```

Comment:
We used NVM in two places:

1. Use nvm to store `P2MPKH` lockup script and `P2SH` unlock script during decoding, so that they can be converted into addresses during the review.
2. Use NVM to store the displayed content during the review

I feel that in most cases 2048 bytes should be enough.